FaunaDB.Client/Types/Value.cs

```diff
@@ -164,16 +164,26 @@ public abstract class Value : Expr
         public static implicit operator Value(DateTimeOffset dt) =>
             FromDateTimeOffset(dt);
 
-        internal static Value FromDateTime(DateTime dt)
+        internal static Value FromDateTime(DateTime dt, Type t = null)
         {
+            if(t == typeof(TimeV))
+                return new TimeV(dt);
+            else if(t == typeof(DateV))
+                return new DateV(dt);
+
             if (dt.Ticks % (24 * 60 * 60 * 10000) > 0)
```

Comment:
instead of the above logic, you can just add a `||` condition on this `if`
```csharp
if(t == typeof(TimeV) || dt.Ticks % (24 * 60 * 60 * 10000) > 0)
```